src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Reindex/ReindexUtilities.cs

```diff
@@ -71,19 +71,12 @@ public async Task ProcessSearchResultsAsync(SearchResult results, IReadOnlyDicti
                 entry.Resource.SearchParameterHash = searchParamHash;
                 var resourceElement = _deserializer.Deserialize(entry.Resource);
                 var newIndices = _searchIndexer.Extract(resourceElement);
-                var newIndicesHash = new HashSet<SearchIndexEntry>(newIndices);
-                var prevIndicesHash = entry.Resource.SearchIndices != null ?
-                    new HashSet<SearchIndexEntry>(entry.Resource.SearchIndices) : new HashSet<SearchIndexEntry>();
 
-                if (newIndicesHash.SetEquals(prevIndicesHash))
-                {
-                    updateHashValueOnly.Add(entry.Resource);
-                }
-                else
-                {
-                    entry.Resource.UpdateSearchIndices(newIndices);
-                    updateSearchIndices.Add(entry.Resource);
-                }
+                // TODO: If it reasonable to do so, we can compare
+                // old and new search indices to avoid unnecessarily updating search indices
+                // when not changes have been made.
+                entry.Resource.UpdateSearchIndices(newIndices);
+                updateSearchIndices.Add(entry.Resource);
```

Comment:
We can remove the call to `UpdateSearchParameterHashBatchAsync` below and also remove `updateHashValueOnly` above